### PR TITLE
dockerfiles: support podman/docker for build.sh

### PR DESCRIPTION
**Pull request by userid0x0**
Allow to force usage of podman/docker by environment variable DOCKER. Example
  export DOCKER=podman
  ./dockerfiles/build.sh
. Try to autodetect podman/docker.

Signed-off-by: Alexander Merkle <alexander.merkle@lauterbach.com>

### CODE REVIEW

File: dockerfiles/build.sh [hunk 1]
```diff
@@ -1,12 +1,17 @@
 #!/bin/sh
 
 set -ex
-
+if [[ -z ${DOCKER} ]]; then
+    command -v podman >/dev/null 2>&1 && DOCKER=podman
+fi
+if [[ -z ${DOCKER} ]]; then
+    command -v docker >/dev/null 2>&1 && DOCKER=docker
+fi
```
Comment (krevsbech) [inline dockerfiles/build.sh:9]: I also favor podman over docker. But in order not to change the default behavior of the scripts consider swapping these two if's

File: dockerfiles/README.rst [hunk 2]
```diff
@@ -17,7 +17,7 @@ Build
 -----
 
 To build one of the above images,
-you need to run the ``docker build`` command in the root of this repository.
+you need to run the ``docker build``  command in the root of this repository.
```
Comment (krevsbech) [inline dockerfiles/README.rst:20]: Why extra "space" ?

**Review by krevsbech — COMMENTED** [review]
Good idea. You are right it is not that elegant to assume docker only. Only minor comments from my side.

**Comment by userid0x0** [comment]
Not a big deal. Better now?

File: dockerfiles/build.sh [hunk 3]
```diff
@@ -1,12 +1,17 @@
 #!/bin/sh
 
 set -ex
-
+if [[ -z ${DOCKER} ]]; then
+    command -v docker >/dev/null 2>&1 && DOCKER=docker
+fi
+if [[ -z ${DOCKER} ]]; then
+    command -v podman >/dev/null 2>&1 && DOCKER=podman
+fi
```
Comment (Bastian-Krause) [inline dockerfiles/build.sh:9]: `[[` won't work if `/bin/sh` is not symlinked to bash. So either adjust the shebang or don't use `[[`.

Comment (userid0x0) [reply dockerfiles/build.sh]: Somewhere back in my brain there rings a bell. It's now `dash` compatible.

**Comment by userid0x0** [comment]
I want to avoid misunderstandings. Let me know if the last push needs another rework.

**Review by jluebbe — APPROVED** [review]

**Review by krevsbech — APPROVED** [review]

**Review by Bastian-Krause — APPROVED** [review]

**Pull request merged**